### PR TITLE
Update framework to .NET Core 2.1

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on ppy/osu-framework#1582.

Pending regression testing.

### CODE REVIEW

**Comment by peppy** [comment]
Looks like file downloads may have regressed (possibly just the progress reporting portion?)

**Comment by peppy** [comment]
Download regression fixed (reference https://github.com/dotnet/corefx/issues/24557#issuecomment-335638904)

File: osu.Game.Rulesets.Mania.Tests/.vscode/launch.json [hunk 1]
```diff
@@ -30,25 +30,25 @@
...
         },
         {
-            "name": "VisualTests (Release, netcoreapp2.0)",
+            "name": "VisualTests (Release, netcoreapp2.1)",
             "type": "coreclr",
             "request": "launch",
             "program": "dotnet",
             "args": [
-                "${workspaceRoot}/bin/Debug/netcoreapp2.0/osu.Game.Rulesets.Mania.Tests.dll"
+                "${workspaceRoot}/bin/Debug/netcoreapp2.1/osu.Game.Rulesets.Mania.Tests.dll"
```
Comment (Tom94) [inline osu.Game.Rulesets.Mania.Tests/.vscode/launch.json]: Can you adjust the path from `Debug` to `Release` here and in all other modes? I know it's a bit unrelated with the PR, but you're already touching the line anyway.

**Review by Tom94 — APPROVED** [review]
👍

**Pull request merged**